coding-concepts.js

```diff
@@ -8,47 +8,76 @@
+// character: "l" index: 2
+// character: "l" index: 3
+// character: "o" index: 4
 
 // --------------------3) What will this log?
 
 const languages = ["JavaScript", "Ruby", "Python", "C++"]
 const index = 1
-// console.log(languages[index])
+console.log(languages[index])
+
+// a) Your answer: "Ruby"
+// b) Verify and explain: "Ruby" is the correct answer. This answer is another example of Zero-indexing but now appling the rule to an array. When looking at an array it is presented as a set of elements: one element being "JavaScript"; when measuring the array's index each element holds a value.
 
-// a) Your answer:
-// b) Verify and explain:
+// For example: 
+// element: "JavaScript"  index: 0
+// element: "Ruby"        index: 1
+// element: "Python"      index: 2
+// element: "C++"         index: 3
 
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
+//output: "ERROR"
+
+// a) Your answer: ["SATURDAY", "SUNDAY"]
+// b) Verify and explain: The output was "ERROR" because .toUpperCase() is a string property and the declared variable holds the data of an array. A string property will not work on an array unless the array is converted into a string with an accessor. Because an accessor doesn't modify the original array I will declare a variable called, joinedArray, and assign its data as the given variable(weekendDays) with the built-in method (.join()) attached. Once the variable is defined I will declare another variable called, upperString, and assign its data as the variable, joinedArray,  with the built-in method (.toUpperCase()) attached. I will then run the upperString variable with the built-in method (.split(" ")) to turn the string into an array. This will output my previous prediction.
+
+// For example:
+var joinedArray = weekendDays.join(" ")
+// console.log(joinedArray.toUpperCase())
+//output: "SATURDAY SUNDAY"
+var upperString = joinedArray.toUpperCase()
+console.log(upperString.split(" "))
+//output: [ 'SATURDAY', 'SUNDAY' ]
+
+// console.log()
 
-// a) Your answer:
-// b) Verify and explain:
 
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
-// console.log(typeof dataTypes.length)
+console.log(typeof dataTypes.length)
+
+// a) Your answer: ERROR?
+// b) Verify and explain: This code will not output anything because inside the argument is an undefined variable "typeof". Im unsure if this is mean't to be a variable or a failed camelcasing? 
 
-// a) Your answer:
-// b) Verify and explain:
+// ...After doing research the output came back as nothing instead of an ERROR because, "The typeof operator returns a string indicating the type of the operand's value.". By using this operator it outputs one six primative operators types. Which the declared variable "dataType" and the length property is not. It didn't come back as underfined because the variable dataType is declared.
```

Comment:
The console.log is asking for 'typeof' (operator) the dataType (variable) .length (built-in method) -- so rather than returning the number 4, it will return the 'typeof' 4 (and 4's datatype is a number).